### ISSUE TITLE
Unable to make WASM version working

### ISSUE THREAD

**Issue by buxx** [not-fl3/macroquad#466]
Hello,

My project started since some years and since recently, my WASM version is failing. There is the browser error : 

```html
[...]
<body>
    <canvas id="glcanvas" tabindex='1'></canvas>
    <script src="[/static/mq_js_bundle.js](view-source:http://127.0.0.1:5000/static/mq_js_bundle.js)"></script>
    <script src="[/static/rollgui2.js](view-source:http://127.0.0.1:5000/static/rollgui2.js)"></script>
    <script>load("/static/engine.wasm");</script>
</body>
[...]
```

![2022-08-11_08-03](https://user-images.githubusercontent.com/1104637/184080605-0ca0347c-48bc-46f4-a111-b00c6814c824.png)

There is the WASM bin dependencies : 

```toml
 [package]
name = "rollgui2"
version = "0.1.0"
edition = "2021"

[profile.dev.package.'*']
opt-level = 3

[dependencies]
# IMPORTANT NOTE : When modifying quad dependencies, think about update the rolling js builder script
macroquad = "0.3"
egui = "0.18.1"
egui-macroquad = { path = "../egui-macroquad" }
quad-storage = "0.1.3"
quad-net = { git = "https://github.com/not-fl3/quad-net", rev = "207da3a4b439ec45d878bde3995af8285940ae20" }
structopt = "0.3.21"
serde = {version="1.0.104", features = ["derive"]}
serde_derive = "1.0.104"
serde_json = "1.0.44"
base64 = "0.13.0"
ureq = "2.4.0"
url = "2.1.1"
ahash = "0.7.6"
image = "0.24.3"

[target.'cfg(target_arch = "wasm32")'.dependencies]
sapp-jsutils = "0.1"

[profile.release]
lto = true
codegen-units = 1
panic = "abort"
```

There is the `mq_js_bundle.js` build script which I use : 

```bash
#!/usr/bin/env bash
set -e

rm -f static/mq_js_bundle.js
curl https://raw.githubusercontent.com/not-fl3/miniquad/8c64a1361eab03469cba223f44eab8bcd177e28c/js/gl.js >> static/mq_js_bundle.js
curl https://raw.githubusercontent.com/not-fl3/quad-snd/f1b04bb32c56073388d25a14bd762d810046ec39/js/audio.js >> static/mq_js_bundle.js
curl https://raw.githubusercontent.com/not-fl3/sapp-jsutils/3dffa01d4cebac58eba37a15755ef20d21ca9d25/js/sapp_jsutils.js >> static/mq_js_bundle.js
curl https://raw.githubusercontent.com/buxx/quad-net/d8e582c5001f977ec42e5f8dd21c1c0bbc9463e7/js/quad-net.js >> static/mq_js_bundle.js
curl https://raw.githubusercontent.com/not-fl3/egui-miniquad/82b6ac065b102c6a9181d4ad73594323e054dee2/docs/quad-url.js >> static/mq_js_bundle.js
curl https://raw.githubusercontent.com/optozorax/quad-storage/3760b953aec17d65cc4ca8edfa39c38e7337ec3a/js/quad-storage.js >> static/mq_js_bundle.js

sed -i 's/js_objects = {}/var js_objects = {}/' static/mq_js_bundle.js
sed -i 's/unique_js_id = 0/var unique_js_id = 0/' static/mq_js_bundle.js
sed -i 's/params_set_mem = function/var params_set_mem = function/' static/mq_js_bundle.js
sed -i 's/params_register_js_plugin = function/var params_register_js_plugin = function/' static/mq_js_bundle.js
```

I try since days to change `mq_js_bundle.js` recipe without success. The message `TypeError: import object field '__wbindgen_placeholder__' is not an Object` remind me the `wasm-bindgen` but I don't use it. Maybe a dependency ? I think about `egui` (and `egui-macroquad`/`egui-miniquad`) but it was working before.

Any idea what's going on ? It is macroquad related ?

**Comment by buxx** [comment]
I just did some investigations and this is the `use image::{io::Reader as ImageReader, DynamicImage};` line which cause the `TypeError: import object field '__wbindgen_placeholder__' is not an Object` error. I suppose some `wasm-bindgem` stuff are done by image crate. I will have to try a replacement ...

**Comment by not-fl3** [comment]
I highly doubt its the `image` responsible for bringing wasm-bindgen, but you can check in `cargo tree`

The problem was discussed a lot here, please search for "wasm-bindgen" keywords in github issues :pray:  
TLDR: if you want to use wasm-bindgen - it is possible with macroquad, but not super easy

**Comment by not-fl3** [comment]
(its probably `ureq`)

**Comment by buxx** [comment]
Hello @not-fl3, after more investigation I confirm `image`crate is the cause. Specially [0.21](https://github.com/image-rs/image/blob/0dfc52dd7629e00262710a49ffbc85ac7d28b38a/CHANGES.md#version-021) version. Using `image` `0.20.1` version remove the `wasm-bindgem` requirement. I opened an issue on `image` repo : https://github.com/image-rs/image/issues/1769

So, this issue is not related to macroquad. I close it and hope it will can help other person in future :)

**Comment by buxx** [comment]
According to https://github.com/image-rs/image/issues/1769#issuecomment-1212834476 you may be right about `ureq` :+1: I will cofirm this.
